org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -100,37 +104,29 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 				}
 			}
 		}
-		startSynchronization(rootFolder.toFile(), distribution, NewProjectHandler.IMPORT_AND_MERGE);
+		return distribution;
 	}
 
-	protected void startSynchronization(File location, GradleDistribution distribution, NewProjectHandler newProjectHandler) {
-		List<String> jvmArgs = new ArrayList<>();
-		IProxyService proxyService = JavaLanguageServerPlugin.getInstance().getProxyService();
-		IProxyData proxy = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PORT, String.valueOf(proxy.getPort()));
-			}
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_USER, proxy.getUserId());
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTP_PROXY_PASSWORD, proxy.getPassword());
-		}
-		proxy = proxyService.getProxyData(IProxyData.HTTPS_PROXY_TYPE);
-		if (proxy != null) {
-			addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_HOST, proxy.getHost());
-			if (proxy.getPort() > 0) {
-				addArgs(jvmArgs, JavaLanguageServerPlugin.HTTPS_PROXY_PORT, String.valueOf(proxy.getPort()));
+	protected void startSynchronization(Path rootFolder, IProgressMonitor monitor) {
+		boolean shouldSynchronize = true;
+		File location = rootFolder.toFile();
+		IProject[] projects = ResourcesPlugin.getWorkspace().getRoot().getProjects();
+		for (IProject project : projects) {
+			File projectDir = project.getLocation() == null ? null : project.getLocation().toFile();
+			if (projectDir != null && projectDir.equals(location)) {
+				PersistentModel model = CorePlugin.modelPersistence().loadModel(project);
+				if (model.isPresent()) {
+					shouldSynchronize = false;
+					break;
```

Comment:
you can break after you found a matching project in all cases

Reply:
I'm not sure I understand that. Could you, please, provide an example?